core/src/modules/locks.ts

```diff
@@ -41,3 +41,35 @@ export async function waitForLock(
 
   return { releaseLock, attempts };
 }
+
+export type getLockResponse = {
+  releaseLock: Function;
+  isLocked: boolean;
+  lockedBy: string;
+};
```

Comment:
Since you aren't re-using this type anywhere, it's usually better to just have the getLock method determine it's own type from the implementation.   This way, there's nothing to change downsteam